astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
 io_registry.register_reader('fits', Table, read_table_fits)
 io_registry.register_writer('fits', Table, write_table_fits)
 io_registry.register_identifier('fits', Table, is_fits)
+
+
+def read_data_fits(filename, ext_data=0, ext_meta=0, ext_mask='mask',
+                   ext_uncert='uncert', kw_unit='bunit', copy=False,
+                   dtype=None, **kwargs_for_open):
+    """
+    Read data from a FITS file and wrap the contents in a \
+    `~astropy.nddata.NDData`.
+
+    Parameters
+    ----------
+    filename : str and other types
+        see :func:`~astropy.io.fits.open` what possible types are allowed.
+
+    ext_data, ext_meta, ext_mask, ext_uncert : str or int, optional
+        Extensions from which to read ``data``, ``meta``, ``mask`` and
+        ``uncertainty``.
+        Default is ``0`` (data), ``0`` (meta), ``'mask'`` (mask) and
+        ``'uncert'`` (uncertainty).
+
+    kw_unit : str or None, optional
+        The header keyword which translates to the unit for the data. Set it
+        to ``None`` if parsing the unit results in a ValueError during reading.
+        Default is ``'bunit'``.
+
+    copy : bool, optional
+        Copy the data while creating the `~astropy.nddata.NDData` instance?
+        Default is ``False``.
+
+    dtype : `numpy.dtype`-like or None, optional
+        If not ``None`` the data array is converted to this dtype before
+        returning. See `numpy.ndarray.astype` for more details.
+        Default is ``None``.
+
+    kwargs_for_open :
+        Additional keyword arguments that are passed to
+        :func:`~astropy.io.fits.open` (not all of them might be possible).
+    """
+
+    # Hardcoded values to get additional information about mask and uncertainty
+    kw_hdr_masktype = 'boolean mask'
+    kw_hdr_uncerttype = {
+        'standard deviation uncertainty': StdDevUncertainty,
+        'unknown uncertainty type': UnknownUncertainty}
+
+    with fits_open(filename, mode='readonly', **kwargs_for_open) as hdus:
+        # Read the data and meta from the specified extensions
+        data = hdus[ext_data].data
+        if dtype is not None:
+            data = data.astype(dtype)
+        meta = hdus[ext_meta].header
+
+        # Read the mask and uncertainty from the specified extensions but
+        # silently fail if the extension does not exist.
+        mask = None
+        if ext_mask in hdus:
+            mask = hdus[ext_mask].data
+            # Convert it to boolean array?
+            if kw_hdr_masktype in hdus[ext_mask].header.get('comment', []):
+                mask = mask.astype(bool)
+
+        uncertainty = None
+        if ext_uncert in hdus:
+            uncertainty = hdus[ext_uncert].data
+
+            hdr = hdus[ext_uncert].header
+            # Get the required class for the uncertainty
+            cls = (kw_hdr_uncerttype[kw] for kw in kw_hdr_uncerttype
+                   if kw in hdr.get('comment', []))
```

Comment:
As above, I think it would be better to have the user specify a class for the uncertainty rather than parsing it from a comment.